airflow/providers/amazon/aws/triggers/rds.py

```diff
@@ -129,16 +130,21 @@ def __init__(
         waiter_max_attempts: int,
         aws_conn_id: str,
         response: dict[str, Any],
-        db_type: RdsDbType,
+        db_type: RdsDbType | str,
```

Comment:
Should we use `Literal` instead? That would make our life easier no? Here, it would be `db_type: Literal["instance", "cluster"]`

Reply:
Yeah but then if we do that it means that the definition of DB types is in multiple places...
And also, I'd have to update usage wherever those triggers are called, while here the fix is transparent.

Reply:
Agree on 2. But on he first point you can define your literal like `db_type = Literal[...]` and then use and import this one across the code

Reply:
hmm, yeah. It might be a bit weird to have 2 different types for the same thing, and also it looks like there is no way to create the literal from the enum https://stackoverflow.com/questions/73491617/convert-enum-to-literal-type-alias-in-python-typing
but I can create it next to the enum, at least it'd be obvious when changing one that the other should be kept in sync.
But again that's a lot of work for an interface that's going to be used only by computers and not users.

Reply:
I am fine with the enum :)